### ISSUE TITLE
Behavior of Go To Definition on Windows

### ISSUE THREAD

**Issue by lmapii** [microsoft/vscode-cpptools#9240]
### Discussed in https://github.com/microsoft/vscode-cpptools/discussions/9238

<div type='discussions-op-text'>

<sup>Originally posted by **lmapii** April 26, 2022</sup>
Version: 1.66.2 (user setup)
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:46:01.075Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 10.0.19044

C/C++ v1.9.8

When using "Go To Definition" on `#define` macros the behavior seems to have changed: On macOS, this command always goes to the first expansion of the definition, i.e., to the actual definition. On windows, this now seems to jump to the full expansion in case of re-definitions without a parenthesis.

Consider the following:
```
module_a.h
    #define MODULE_A_SMTH 0x3
module_b.h
    #define MODULE_B_SMTH MODULE_A_SMTH
module_b.c
    uint8_t some_variable = MODULE_B_SMTH;
```

When using the context menu (or CTRL/ALT+click) on `MODULE_B_SMTH` within `module_b.c`, vscode now jumps to the definition of `MODULE_A_SMTH` instead of `MODULE_B_SMTH`.

I'd expect it to always go to the actual definition without expanding the macro - I can always continue navigating if it is not expanded. Added files and a screen recording.

Is there maybe some configuration option that has been added or am I missing something?

[pkg.zip](https://github.com/microsoft/vscode-cpptools/files/8561052/pkg.zip)

https://user-images.githubusercontent.com/7037131/165251590-d7d35abe-bfed-45ec-96c0-5e22c0bd8efd.mp4

</div>

**Comment by Colengms** [comment]
Hi @lmapii .  What IntelliSenseMode you are using?  Are you configured to use cl.exe or a version of gcc or clang?  (The output of the `C/C++: Log Diagnostics` command could answer these questions).

Also, could you try installing the latest Pre-Release build of the extension (currently 1.10.0) to see if the issue still repros?

This looks very similar to https://github.com/microsoft/vscode-cpptools/issues/8914 , which is actually the same issue as https://github.com/microsoft/vscode-cpptools/issues/8577 . Both should be fixed in 1.10.0.  However, these involve using a gcc/clang IntelliSenseMode.  If you are using `cl.exe` on Windows, you may be seeing a different issue.

**Comment by lmapii** [comment]
Thank you for the very fast feedback! This is the output of the diagnostics.

```
-------- Diagnostics - 27.4.2022, 08:08:44
Version: 1.9.8
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "compilerPath": "C:\\MinGW\\bin\\gcc.exe",
    "intelliSenseMode": "windows-gcc-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "defines": [
        "_MAKE_CORE_COUNT_=6"
    ],
    "compilerArgs": [],
    "cStandard": "c99",
    "cppStandard": "c++11",
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
No active translation units.
```

So I'm using `gcc`. I've installed the Pre-Release of the extension and indeed it seems to work fine there :)
